### PR TITLE
Update VF Profile Page

### PR DESCRIPTION
**Pull request by jonboiser**
1. Adds (back) the password reset section with validation
1. Removes the change username section, since VF users are IDd by full name
1. Move the submit button

## Password Validation
password validation

## Fullname Validation
name validation

### CODE REVIEW

File: kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue [hunk 1]
```diff
@@ -30,18 +30,10 @@
     </section>
 
     <form @submit.prevent="submitEdits">
-      <ui-alert
```
Comment (christianmemije) [inline kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue]: Hmm, @indirectlylit had mentioned that we should try to avoid unnecessary changes.

File: kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue [hunk 2]
```diff
@@ -295,11 +287,8 @@
     margin-right: auto
     width: ($iphone-width - 20)px
 
-  .submit
-    margin-left: auto
-    margin-right: auto
-    display: block
-
+  button[type='submit']
```
Comment (christianmemije) [inline kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue]: What should do for changes that should also be part of the main user plugin, like this change?

File: kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue [hunk 3]
```diff
@@ -135,10 +138,12 @@
       points: 'Points',
       role: 'Role',
       devicePermissions: 'Device permissions',
-      usernameNotAlphaNumUnderscore: 'Username can only contain letters, numbers, and underscores',
       required: 'This field is required',
       limitedPermissions: 'Limited permissions',
       youCan: 'You can',
+      newPw: 'New password',
+      newPwConfirm: 'New password again',
```
Comment (christianmemije) [inline kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue]: Could we change to `Confirm new password`?

File: kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue [hunk 4]
```diff
@@ -177,52 +185,41 @@
...
-        }
-        return '';
+      canEditPassword() {
+        return this.userIsAdmin || this.facilityConfig.learnerCanEditPassword;
       },
-      nameIsInvalid() {
-        return !!this.nameIsInvalidText;
+      // i.e. not a coach or a learner
+      userIsAdmin() {
+        return !this.isCoach && !this.isLearner;
```
Comment (christianmemije) [inline kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue]: So superuser learners and coaches cannot?

File: kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue [hunk 5]
```diff
@@ -177,52 +185,41 @@
...
       },
-      nameIsInvalidText() {
-        if (this.nameBlurred || this.formSubmitted) {
-          if (this.name === '') {
-            return this.$tr('required');
-          }
-        }
-        return '';
+      canEditPassword() {
+        return this.userIsAdmin || this.facilityConfig.learnerCanEditPassword;
```
Comment (christianmemije) [inline kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue]: So superuser learners and coaches cannot edit their password?

File: kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue [hunk 6]
```diff
@@ -177,52 +185,41 @@
...
-          if (this.username === '') {
-            return this.$tr('required');
-          }
-          if (!validateUsername(this.username)) {
-            return this.$tr('usernameNotAlphaNumUnderscore');
-          }
-        }
-        return '';
+      nameIsValid() {
+        return this.name.trim() !== '';
```
Comment (christianmemije) [inline kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue]: The k-textbox already trims  https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/views/k-textbox/index.vue#L7

File: kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue (hunk 5 above)
Comment (rtibbles) [reply kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue]: As there should not be too many superusers floating about in the VF use case, we can probably defer that question for consideration on Kolibri - open an issue on Kolibri about it?

Comment (jonboiser) [reply kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue]: This page might not have been updated after the superuser permissions stuff. To me, `userIsAdmin` might use the `isAdmin` getter, which looks for the superuser role.

File: kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue [hunk 7]
```diff
@@ -150,23 +155,26 @@
...
         formSubmitted: false,
+        newPw: '',
+        newPwBlurred: false,
+        newPwConfirm: '',
+        newPwConfirmBlurred: false,
       };
     },
     computed: {
       role() {
-        if (this.getUserRole === UserKinds.ADMIN) {
```
Comment (rtibbles) [inline kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue]: This is the kind of refactor that @indirectlylit was requesting that we try to avoid - what is the motivation for changing it from an if sequence to a switch?

Comment (jonboiser) [reply kolibri_instant_schools_plugin/assets/src/views/profile-page/index.vue]: DRYness and readability. I will revert this.
